### PR TITLE
fix: Ignore surrounding whitespace for `LineBreak` in `XamlReader` and Xaml generator

### LINKED ISSUE #11292: [iOS][Android] A `Xaml.Documents.LineBreak` produces a slight leading offset compared to regularly wrapped Text.
### Current behavior

The following text
```
<TextBlock TextWrapping="WrapWholeWords">
   LineBreak
<LineBreak />
   You can construct URLs and access their parts. For URLs that represent local files, you can also manipulate properties of those files directly.
<LineBreak />
   LineBreak 
</TextBlock>
```
is output like this on Windows & mobile devices respectively.

Screenshot 2023-02-13 124128

Screenshot 2023-02-13 123729

Note the apparent white-space leading the 2nd and 5th line (each produced by a `LineBreak`).

### Expected behavior

Line beginnings should be succinct.

### How to reproduce it (as minimally and precisely as possible)

See above

### Workaround

_No response_

### Works on UWP/WinUI

Yes

### Environment

Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia

### NuGet package version(s)

Uno.WinUI 4.8.0-dev.299

### Affected platforms

Android, iOS

### IDE

_No response_

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): closes #11292

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Comment by Youssef1313** [comment]
Looks like my assumption was wrong.

For the following code:

```xml
        <TextBlock x:Name="myTb1">Hello <Italic>World</Italic></TextBlock>
        <TextBlock x:Name="myTb2">Hello <LineBreak /> world</TextBlock>
```

I get the following on Windows:

image

File: src/Uno.UI/UI/Xaml/Markup/Reader/XamlStringParser.cs [hunk 1]
```diff
@@ -140,25 +141,42 @@ private XamlMemberDefinition VisitMember(XamlXmlReader reader, XamlObjectDefinit
...
+							lastWasTrimSurroundingWhiteSpace = false;
 							member.Value = reader.Value;
 						}
 						break;
 
 					case XamlNodeType.StartObject:
 						_depth++;
-						member.Objects.Add(VisitObject(reader, owner));
+						var obj = VisitObject(reader, owner);
+						if (!reader.PreserveWhitespace && lastWasLiteralInline && obj.Type.TrimSurroundingWhitespace && member.Objects.Count > 0 &&
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Markup/Reader/XamlStringParser.cs:160]: Let's align each condition on a new line

File: src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileParser.cs [hunk 2]
```diff
@@ -519,14 +524,27 @@ private XamlMemberDefinition VisitMember(XamlXmlReader reader, XamlObjectDefinit
 
 					case XamlNodeType.StartObject:
 						_depth++;
-						member.Objects.Add(VisitObject(reader, owner));
+						var obj = VisitObject(reader, owner);
+						if (!reader.PreserveWhitespace && lastWasLiteralInline && obj.Type.TrimSurroundingWhitespace && member.Objects.Count > 0 &&
```
Comment (jeromelaban) [inline src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileParser.cs:528]: Let's align conditions on new lines

**Review by jeromelaban — APPROVED** [review]

File: src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileParser.cs (hunk 2 above)
Comment (Youssef1313) [inline src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileParser.cs:528]:
```suggestion
						if (!reader.PreserveWhitespace &&
							lastWasLiteralInline &&
							obj.Type.TrimSurroundingWhitespace &&
							member.Objects.Count > 0 &&
```

File: src/Uno.UI/UI/Xaml/Markup/Reader/XamlStringParser.cs (hunk 1 above)
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Markup/Reader/XamlStringParser.cs:160]:
```suggestion
						if (!reader.PreserveWhitespace &&
							lastWasLiteralInline &&
							obj.Type.TrimSurroundingWhitespace &&
							member.Objects.Count > 0 &&
```

**Pull request merged**